tracer/src/Datadog.Trace/Agent/Transports/HttpClientRequest.cs

```diff
@@ -64,12 +63,12 @@ public async Task<IApiResponse> PostAsJsonAsync(IEvent events, JsonSerializer se
             }
         }
 
-        public async Task<IApiResponse> PostAsync(ArraySegment<byte> traces)
+        public async Task<IApiResponse> PostAsync(ArraySegment<byte> traces, string contentType)
         {
             // re-create HttpContent on every retry because some versions of HttpClient always dispose of it, so we can't reuse.
             using (var content = new ByteArrayContent(traces.Array, traces.Offset, traces.Count))
             {
-                content.Headers.ContentType = new MediaTypeHeaderValue("application/msgpack");
+                content.Headers.ContentType = new MediaTypeHeaderValue(contentType);
```

Comment:
Can we reuse this instance?

Reply:
I expect so, but given the comment above I'd rather we looked into that in a separate PR, in case it introduces issues?

Reply:
Totally agree

Reply:
See https://github.com/dotnet/corefx/pull/19082

The old (bad) behavior (HttpClient disposes HttpContent) will not change in .NET Framework's built-in `HttpClient ` because it is a breaking change.

Reply:
> See [dotnet/corefx#19082](https://github.com/dotnet/corefx/pull/19082)
> 
> The old (bad) behavior (HttpClient disposes HttpContent) will not change in .NET Framework's built-in `HttpClient ` because it is a breaking change.

Does that preclude us caching the `MediaTypeHeaderValue` though?

Reply:
I'm not sure `MediaTypeHeaderValue` gets disposed. If that is the case, we can still avoid the string parsing in the constructor by calling `IClonable.Clone` of the [instance](https://source.dot.net/#System.Net.Http/System/Net/Http/Headers/MediaTypeHeaderValue.cs,256).

Reply:
Given `MediaTypeHeaderValue` is not disposable, seems safe to reuse it to me?

Reply:
True!